### PR TITLE
Compress stored data

### PR DESCRIPTION
**Pull request by franziska-warncke**
use alternative data compression tools to gzip #162 

EDIT: After review, we decided to change this to an other topic: **Compress stored data**

### CODE REVIEW

**Comment by markus-ntt-seidl** [comment]
@franziska-hesselfeld-nttdata @aoifefitton (referencing you as the original author, but GH doesn't seem to know you ...?)
I'm a bit opposed to the fact that we recommend a specific format. The main reason for this is, that we do not know the use case and architectural requirements of the software using it. Is it permanent storage for archival (legal) purposes, where the probability that it will ever be read is low?
We had a few similar in the Web section, for example https://patterns.greensoftware.foundation/catalog/web/serve-images-in-modern-formats or https://patterns.greensoftware.foundation/catalog/web/deprecate-gifs . Even with the architectural requirement (Web) we also highlighted other formats (or hinted about it).

---- 
Just saw that this is web, only (or was cloud meant?). So this was covered already: https://github.com/Green-Software-Foundation/green-software-patterns/pull/90 (still in SME though)

**Comment by franziska-warncke** [comment]
@markus-gsf-seidl 
Somehow I understand, that it is the same topic (compression), but somehow it is a bit different. So in #90, the pattern suggest to use compression in general. In this pattern, we suggest (or should suggest) to use a compression tool which fits to the right use case. So in the blog post, the writer pointed out, that the usage of different compression tool as gzip can lead to a reduction in storage and therefore compression time.  So not only choosing compression itself, but which compression tool can have an effect on the reduction of energy. 
My suggestion:
Maybe we can integrate this information as part of the other pattern what we already have, so when reading through the pattern, we also have to have in mind choosing the right compression tool.
Whats your opinion on that?

**Comment by markus-gsf-seidl** [comment]
@franziska-warncke 

Looking at this after a few months, it may make sense to have a different pattern. Maybe we leave this in cloud (and make it storage related?) and leave the #90 in web as it is. There is a difference, in web client and server must agree on a compression they both support. Whereas in storage you are (mostly) free to choose the a specific compression tool.

I would prefer this pattern to be worded to use a good tool for your use case and not a specific algorithm like ZSTD.

**Comment by franziska-warncke** [comment]
@markus-gsf-seidl 

Sounds reasonable and I like this idea. I will look up, if I can find resources for this kind of pattern and will change that up.

**Comment by franziska-warncke** [comment]
@markus-gsf-seidl Please have a look again.

File: docs/catalog/cloud/compress-stored-data.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+Use the right compression tool for the use case to reduce the storage requirements, including both capacity and required bandwidth to write or retrieve data. 
+
+## SCI Impact
+
+`SCI = (E * I) + M per R`  
+[Software Carbon Intensity Spec](https://grnsft.org/sci)
+
+Concerning the SCI equation, using compression when storing data impact one part:
+
+- `E`: Decrease the amount of storage and therefor less E for the storage. However, we should be aware that there may be a slight increase in energy consumed due to compressing and de-compressing data.
```
Comment (markus-gsf-seidl) [inline docs/catalog/cloud/compress-stored-data.md:28]: The "However" part would be best when you extract it into the "I"ntensity. :)

File: docs/catalog/cloud/compress-stored-data.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+## Assumptions
+
```
Comment (markus-gsf-seidl) [inline docs/catalog/cloud/compress-stored-data.md:31]: One assumption could be, that you have the ability to choose. If you store much, much data you are forced to use compression to keep storage costs reasonable.
Another one: If you are storage space constrainted you are force to use a high compression (even when there is high CPU required) and also the other way around.

**Review by markus-gsf-seidl — CHANGES_REQUESTED** [review]
Minor review comments / extensions.

File: docs/catalog/cloud/compress-stored-data.md (hunk 1 above)
Comment (franziska-warncke) [reply docs/catalog/cloud/compress-stored-data.md:28]: Can you give me are more detailed explanation, why this is part of the Intensity?

Comment (markus-gsf-seidl) [reply docs/catalog/cloud/compress-stored-data.md:28]: You are right, it should be Emboddied CO2 and Energy CO2, which are E and M

**Comment by franziska-warncke** [comment]
I think it is ready for SME Review. @greenhsu123 Can you have a look?

**Review by markus-gsf-seidl — APPROVED** [review]
LGTM

**Pull request merged**